homeassistant/components/notify/pushsafer.py

```diff
@@ -27,21 +56,125 @@
+
         title = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
-        payload = {'k': self._private_key, 't': title, 'm': message}
-        response = requests.get(_RESOURCE, params=payload,
-                                timeout=DEFAULT_TIMEOUT)
-        if response.status_code != 200:
-            _LOGGER.error("Not possible to send notification")
+        data = kwargs.get(ATTR_DATA)
+
+        if data is None:
+            data = {}
+        # Converting the specified image to base64
+        picture1 = data.get(ATTR_PICTURE1)
+        picture1_encoded = ""
+        if picture1 is not None:
+            _LOGGER.debug("picture1 is available")
+            url = picture1.get(ATTR_PICTURE1_URL, None)
+            local_path = picture1.get(ATTR_PICTURE1_PATH, None)
+            username = picture1.get(ATTR_PICTURE1_USERNAME)
+            password = picture1.get(ATTR_PICTURE1_PASSWORD)
+            auth = picture1.get(ATTR_PICTURE1_AUTH)
+
+            if url is not None:
+                _LOGGER.debug("Loading image from url %s", url)
+                picture1_encoded = self.loadfromurl(url, username, password,
+                                                    auth)
+            elif local_path is not None:
+                _LOGGER.debug("Loading image from file %s", local_path)
+                picture1_encoded = self.loadfromfile(local_path)
+            else:
+                _LOGGER.warning("missing url or local_path for picture1")
+        else:
+            _LOGGER.debug("picture1 is not specified")
+
+        payload = {
+            'k': self._private_key,
+            't': title,
+            'm': message,
+            's': data.get(ATTR_SOUND, ATTR_SOUND_DEFAULT),
+            'v': data.get(ATTR_VIBRATION, ATTR_VIBRATION_DEFAULT),
+            'i': data.get(ATTR_ICON, ATTR_ICON_DEFAULT),
+            'c': data.get(ATTR_ICONCOLOR, ATTR_ICONCOLOR_DEFAULT),
+            'u': data.get(ATTR_URL, ATTR_URL_DEFAULT),
+            'ut': data.get(ATTR_URLTITLE, ATTR_URLTITLE_DEFAULT),
+            'l': data.get(ATTR_TIME2LIVE, ATTR_TIME2LIVE_DEFAULT),
+            'p': picture1_encoded
+        }
+
+        _LOGGER.debug("using push data: %s", str(payload))
+
+        for target in targets:
+            payload['d'] = target
+            response = requests.post(_RESOURCE, data=payload,
+                                     timeout=CONF_TIMEOUT)
+            if response.status_code != 200:
+                _LOGGER.error("Pushsafer failed with: %s", response.text)
+            else:
+                _LOGGER.debug("Push send: %s", response.json())
+
+    @classmethod
+    def getbase64(cls, filebyte, mimetype):
+        """Convert the image to the expected base64 string of pushsafer."""
+        if mimetype not in _ALLOWED_IMAGES:
+            _LOGGER.warning("%s is a not supported mimetype for images",
+                            mimetype)
+            return None
+        else:
```

Comment:
Not necessary else.

Reply:
You can remove this else since you return above.

Reply:
its needed
if mimetype is not in _ALLOWED_IMAGES return NONE otherwise return the bases64 encoded data string

Reply:
No, you can remove the `else` and indent out the code below it.